src/engine/renderer/gl_shader.h

```diff
@@ -589,7 +589,7 @@ class GLUniform4f : protected GLUniform
 #if defined( USE_UNIFORM_FIREWALL )
 		vec4_t *firewall = ( vec4_t * ) &p->uniformFirewall[ _firewallIndex ];
 
-		if ( Vector4Compare( *firewall, v ) )
+		if ( !memcmp( *firewall, v, sizeof( v ) ) )
```

Comment:
Thanks to the wacky C behavior with arrays as function arguments, they type of `v` is `float*` so its size will be pointer size. Use `sizeof( *firewall )` instead.

Alternatively you could use the Math::Vector lib we have. Like `Vec4::Load(firewall) == Vec4::Load(v)`

Reply:
You didn't fix it

Reply:
woopsie. Got messed in my rebases.